pep-0706.rst

```diff
@@ -0,0 +1,509 @@
+
+Other than that, *all* of the changes to ``tarfile`` should be backported, so
+``hasattr(tarfile, 'data_filter')`` becomes a reliable check for all
+of the new functionality.
+
+Note that CPython's usual policy is to avoid adding new API in security
+backports.
+This feature does not make sense without new API
+(``TarFile.extraction_filter`` and the ``filter`` argument),
+so we'll make an exception.
+(See `Discourse comment 23149/16 <https://discuss.python.org/t/23149/16?u=encukou>`__
+for details.)
+
+Here are examples of code that takes into account that ``tarfile`` may or may
+not not have the proposed feature.
+
+When copying these snippets, note that setting ``extraction_filter``,
+will affect subsequent operations.
+
+* Fully trusted archive::
+
+    my_tarfile.extraction_filter = (lambda member: member)
+    my_tarfile.extractall()
+
+* Use the ``'data'`` filter if available, but revert to Python 3.11 behavior
+  (``'fully_trusted'``) if this feature is not available::
+
+    my_tarfile.extraction_filter = getattr(tarfile, 'data_filter', lambda x: x)
+    my_tarfile.extractall()
+
+  (This is an unsafe operation, so it should be spelled out explicitly,
+  ideally with a comment.)
+
+* Use the ``'data'`` filter, *fail* if it is not available::
+
+    my_tarfile.extractall(filter=tarfile.data_filter)
+
+  or::
+
+    my_tarfile.extraction_filter = tarfile.data_filter
+    my_tarfile.extractall()
+
+* Use the ``'data'`` filter, *warn* if it is not available::
+
+   if hasattr(tarfile, 'data_filter'):
+       my_tarfile.extractall(filter='data')
+   else:
+       # remove this when no longer needed
+       warn('Extracting may be unsafe, consider updating Python')
+       my_tarfile.extractall()
+
+
+Security Implications
+=====================
+
+This proposal improves security, at the expense of backwards compatibility.
+In particular, it will help users avoid `CVE-2007-4559`_.
+
+
+How to Teach This
+=================
+
+The API, usage notes and tips for further verification will be added to
+the documentation.
+These should be usable for users that know extracting an archive, but are
+not familiar with the specifics of UNIX filesystems nor the related security
+issues.
+
+
+Reference Implementation
+========================
+
+XXX None yet.
+
+
+Rejected Ideas
+==============
+
+SafeTarFile
+-----------
+
+An initial idea from Lars Gustäbel was to provide a separate class that
+implements security checks (see `gh-65308`_).
+There are two major issues with this approach:
+
+* The name is misleading. General achive operations can never be made “safe”
+  from all kinds of unwanted behavior, without impacting legitimate use cases.
+* It does not solve the problem of unsafe defaults.
+
+However, many of the ideas behind SafeTarFile were reused in this PEP.
+
+Add absolute_path option to tarfile
+-----------------------------------
+
+A minimal change to check the “CVE resolved” box doesn't go far enough to
```

Comment:
Perhaps worth being slightly more explicit about what such a proposed option was intended to do (I'm assuming something related to disallowing tar files with absolute paths, but what specifically?) And I'm not sure what "option" is intended to mean in this context—is this module data? A class attribute? A function/method parameter? To what, specifically?